### ISSUE TITLE
OntologyDesign sprint: iteration 2 (similarities)

### ISSUE THREAD

**Issue by andreamust** [polifonia-project/sonar2021_demo#21]
As a second iteration in modelling the ontologies for the Sonar demo we are focusing on similarity. In particular, starting from the requirements of the demo it became necessary to model two different types of similarity:

- _Harmonic Similarity_ [computed by @jonnybluesman and @andreamust, and available [here](https://github.com/polifonia-project/sonar2021_demo/tree/datasets/harmonic-sim) - in progress]
- _Lyrics Similarity_ [computed by @roccotrip, and available [here](https://github.com/polifonia-project/lyrics_similarity) - in progress]

To represent these two different types of similarity we have sketched two models.

# Harmonic Similarity
![harmonicSimilarity](https://user-images.githubusercontent.com/44606182/134485104-8539ebe7-63a8-45ca-a17b-99f7a82895b6.png)

### Issues on Harmonic Similarity
- Will harmonic similarity only be computed between two different sequences?
- Are the labels clear, expressive and formally correct?
- Is it necessary to define binary projections (e.g. for time stamps)?
- What is the right module for this type of similarity?

# Lyrics Similarity
![lyricsSimilarity](https://user-images.githubusercontent.com/44606182/134485995-4ba92f90-6e69-4061-8e03-94e93e63e94e.png)

### Issues on Lyrics Similarity
- Are the labels clear, expressive and formally correct?
- What is the right module for this type of similarity (lyrics should belong to the [composition module](https://liveunibo.sharepoint.com/:p:/r/sites/polifonia/_layouts/15/Doc.aspx?sourcedoc=%7BF5587668-4A79-43C6-9F7E-3195C02E9878%7D&file=2021_07_09_ontology_network.pptx&action=edit&mobileredirect=true), what about the similarity between lyrics)?
- Do we need one more granularity level on "LyricsPart" (e.g. lyrics words)?
- Is the encoding used for embedding negligible for our purposes (see Harmonic Similarity model)?
- How will we compute the position of a LyricsPart within the whole lyrics?

**Comment by valecarriero** [comment]
I think that LyricsSimilarity can be in the musical composition module. Don't know about the general concept of Similarity.

**Comment by andreamust** [comment]
> I think that LyricsSimilarity can be in the musical composition module. Don't know about the general concept of Similarity.

I agree, I have changed the prefix of LyricSimilarity in the diagram. As for the concept of similarity, it remains an open point.

**Comment by andreamust** [comment]
![harmonicSimilarity_data](https://user-images.githubusercontent.com/44606182/134688933-8386dc8d-1168-42d6-84e6-9f4ab78c78b4.png)

I add a few examples of data related to Harmonic Similarity, to help understand what types of data we will be dealing with.

**Comment by vpresutti** [comment]
> > I think that LyricsSimilarity can be in the musical composition module. Don't know about the general concept of Similarity.
> 
> I agree, I have changed the prefix of LyricSimilarity in the diagram. As for the concept of similarity, it remains an open point.

The concept of two compositions/tracks being similar based on some criteria is a key one to Polifonia. In my opinion the general Similarity pattern and its specialisations should go into a Core module. Wdyt? Another option is to have a general concept of Similarity between two cultural entities (which can be compositions but also of different types) in the Core module and then have specialisations in different mldules. But personally I think that this second option is less manageable and usable. A third option is to have a module dedicated to Comparative measures or something like that, where we define patterns describing observations and measures applied to comparative analysis, such as similarities.
